### ISSUE TITLE
Separate SDK into its own package

### ISSUE THREAD

**Issue by hdost** [open-telemetry/opentelemetry-rust#721]
Based on the Usage Guidelines in https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/library-guidelines.md does it make sense to split the SDK out into its own crate? 
I feel like with the current implementation it kinda already achieves the noop and since unused code won't be in the final binary I suppose it doesn't matter all that much.

**Comment by jtescher** [comment]
Yeah could be nice to split out the api and sdk into their own crates. @GregBowyer would you be open to giving this project access to https://crates.io/crates/opentelemetry-ap and https://crates.io/crates/opentelemetry-sdk?

**Comment by GregBowyer** [comment]
Yes

**Comment by hdost** [comment]
I would just assume keep `opentelemetry` for the API segment and `opentelemetry-sdk` be the SDK. Follows kinda the go implementation a bit https://github.com/open-telemetry/opentelemetry-go/blob/main/example/prometheus/main.go not sure if there's an advantage other than maybe delineating between the two.

**Comment by GregBowyer** [comment]
So how do we do this, other than I would like to confirm with a gpg signature whats the policy on the crates.io side

**Comment by jtescher** [comment]
@GregBowyer Not sure what their transfer method is, but I think you can add a github group like @open-telemetry/rust-maintainers to the crate owners list, or if that is a hassle just add @djc, @TommyCpp and myself 🙏

**Comment by GregBowyer** [comment]
Done:

Added: 
- jtescher
- djc
- tommycpp

**Comment by djc** [comment]
@GregBowyer thanks!

**Comment by TommyCpp** [comment]
@GregBowyer Thanks a lot!

**Comment by djc** [comment]
I also like the idea of using the opentelemetry crate for the API. Don't know what they do in other languages, though?

**Comment by jtescher** [comment]
Took a quick look at splitting things, will help but there are some breaking changes necessary as some of the api trats depend on types currently in the sdk (`InstrumentationLibrary`, `Resource`, possibly `SpanData`, etc).

Also there is a choice to be made on where "sdk api" traits (e.g. `SpanExporter`, `IdGenerator`, `TraceRuntime`) would go. If we keep them in the sdk crate like most of the other otel languages do, then potential alternative sdk implementations will need to depend on the sdk directly to be able to be interoperable with the exporter ecosystem. Alternatively we could move them into the API crate somewhere, but that would deviate a bit from otel expectations.

**Comment by hdost** [comment]
>  If we keep them in the sdk crate like most of the other otel languages do, then potential alternative sdk implementations will need to depend on the sdk directly to be able to be interoperable with the exporter ecosystem.
100% agree, not necessarily sure if that's an issue. what would the downsides be with that?

**Comment by hdost** [comment]
> Took a quick look at splitting things, will help but there are some breaking changes necessary as some of the api trats depend on types currently in the sdk (`InstrumentationLibrary`, `Resource`, possibly `SpanData`, etc).

For the SpanData it looks like there's a Noop Span Exporter, which doesn't really need to exist and then also a testing span exporter, which if we follow others again might even be it's own package which would rely on SDK. However it doesn't need to exist in the API either.

**Comment by TommyCpp** [comment]
Generally speaking. We can build `opentelemetry-api` and `opentelemtry-sdk` and use `opentelemetry` as all in one, battery included solution including both API and SDK. I think the general idea can be to provide a ready-to-use solution while giving people the choice to dig into the code and replace the component as needed.

**Comment by TommyCpp** [comment]
> If we keep them in the sdk crate like most of the other otel languages do, then potential alternative sdk implementations will need to depend on the sdk directly to be able to be interoperable with the exporter ecosystem

I think if the users just want to custom exporter then it's OK for them to rely on SDK create.

**Comment by hdost** [comment]
Thank you @jtescher :)
